File: AlternativeQL/src/org/ql/ast/literal/IntegerLiteral.java
package org.ql.ast.literal;

import org.ql.ast.Node;

public class IntegerLiteral implements Node {
    private int integerLiteral;

    public IntegerLiteral(int integerLiteral) {
        this.integerLiteral = integerLiteral;
    }

    public int getIntegerLiteral() {
        return integerLiteral;
    }
}

File: koopmans/lib/ql/ast/literal/integer_literal.rb
module QL
  module AST
    class IntegerLiteral < Literal
      def initialize(value)
        @value = value.to_i
      end

      def to_value
        value.to_i
      end

      def to_type
        IntegerType.new
      end
    end
  end
end
